src/main/java/org/optaplanner/springboottaskassigning/TaskAssigningSolverManagerController.java

```diff
@@ -16,42 +16,68 @@
 
 package org.optaplanner.springboottaskassigning;
 
+import java.util.NoSuchElementException;
+
 import org.optaplanner.core.api.score.Score;
 import org.optaplanner.springboottaskassigning.domain.TaskAssigningSolution;
-import org.optaplanner.springboottaskassigning.solver.SolverManager;
 import org.optaplanner.springboottaskassigning.solver.SolverStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
 
 @RestController
-@RequestMapping("solvers")
+@RequestMapping("/tenants/{tenantId}/solver")
 public class TaskAssigningSolverManagerController {
 
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
     @Autowired
-    private SolverManager<TaskAssigningSolution> solverManager;
+    private TaskAssigningSolverManagerService solverManagerService;
```

Comment:
In the long run, we want to go to SolverManager<TaskAssigningSolution>, so it's a resuable SolverManager.